services-camel/src/main/java/edu/unc/lib/dl/services/camel/longleaf/LongleafRouter.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.services.camel.longleaf;
+
+import org.apache.camel.BeanInject;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+
+/**
+ * Router for longleaf operations
+ *
+ * @author bbpennel
+ */
+public class LongleafRouter extends RouteBuilder {
+
+    @BeanInject(value = "registerLongleafProcessor")
+    private RegisterToLongleafProcessor registerProcessor;
+
+    @Override
+    public void configure() throws Exception {
+        from("direct-vm:filter.longleaf")
+            .filter().method(RegisterToLongleafProcessor.class, "registerableBinary")
+            .log(LoggingLevel.DEBUG, "Queuing ${headers[CamelFcrepoUri]} for registration to longleaf")
+            .to("sjms:register.longleaf?transacted=true");
```

Comment:
How is this getting to the next route, the only from() goes to sjms-batch?

Reply:
its a quirk of how sjms-batch works, you have to set destinations as `sjms:`, but the route configuration has to be setup with `sjms-batch`. You can see it scattered around this document:
https://camel.apache.org/components/latest/sjms-batch-component.html
My impression is that from the outside, it just looks like an sjms queue, while internally it is configured to do the batching (there might even be two routes behind the scenes)